packages/swapkit/thorchain/src/mayaPlugin.ts

```diff
@@ -42,6 +57,149 @@ const plugin = ({ wallets, stagenet = false }: { wallets: Wallet; stagenet?: boo
+    if (mimir.HALTCHAINGLOBAL >= 1 || mimir.HALTTHORCHAIN >= 1) {
+      throw new SwapKitError("core_chain_halted");
+    }
+
+    return deposit({ assetValue, recipient: "", memo });
+  }
+
+  async function depositToPool({
+    assetValue,
+    memo,
+    feeOptionKey = FeeOption.Fast,
+  }: { assetValue: AssetValue; memo: string; feeOptionKey?: FeeOption }) {
+    const {
+      gas_rate = "0",
+      router,
+      address: poolAddress,
+    } = await getInboundDataByChain(assetValue.chain);
+
+    return deposit({
+      assetValue,
+      recipient: poolAddress,
+      memo,
+      router,
+      feeRate: Number.parseInt(gas_rate) * gasFeeMultiplier[feeOptionKey],
+    });
+  }
+
+  function registerMAYAName({
+    assetValue,
+    ...param
+  }: NameRegisterParam & { assetValue: AssetValue }) {
+    return depositToProtocol({ assetValue, memo: getMemoFor(MemoType.NAME_REGISTER, param) });
+  }
+
+  function nodeAction({ type, assetValue, address }: NodeActionParams) {
+    const memoType =
+      type === "bond" ? MemoType.BOND : type === "unbond" ? MemoType.UNBOND : MemoType.LEAVE;
+    const memo = getMemoFor(memoType, {
+      address,
+      unbondAmount: type === "unbond" ? assetValue.getBaseValue("number") : undefined,
+    });
+    const assetToTransfer = type === "bond" ? assetValue : getMinAmountByChain(Chain.Maya);
+
+    return depositToProtocol({ memo, assetValue: assetToTransfer });
+  }
+
+  function addLiquidityPart({
+    assetValue,
+    poolAddress,
+    address,
+    symmetric,
+  }: AddLiquidityPartParams) {
+    if (symmetric && !address) {
+      throw new SwapKitError("core_transaction_add_liquidity_invalid_params");
+    }
+    const memo = getMemoFor(MemoType.DEPOSIT, {
+      chain: poolAddress.split(".")[0] as Chain,
+      symbol: poolAddress.split(".")[1] as string,
+      address: symmetric ? address : "",
+    });
+
+    return depositToPool({ assetValue, memo });
+  }
+
+  // biome-ignore lint/complexity/noExcessiveCognitiveComplexity: TODO: Refactor
+  async function addLiquidity({
+    runeAssetValue,
+    assetValue,
+    runeAddr,
+    assetAddr,
+    isPendingSymmAsset,
+    mode = "sym",
+  }: AddLiquidityParams) {
+    const { chain, symbol } = assetValue;
+    const isSym = mode === "sym";
+    const runeTransfer = runeAssetValue?.gt(0) && (isSym || mode === "rune");
+    const assetTransfer = assetValue?.gt(0) && (isSym || mode === "asset");
+    const includeRuneAddress = isPendingSymmAsset || runeTransfer;
+    const runeAddress = includeRuneAddress ? runeAddr || getAddress(wallets, Chain.THORChain) : "";
+    const assetAddress = isSym || mode === "asset" ? assetAddr || getAddress(wallets, chain) : "";
+
+    if (!(runeTransfer || assetTransfer)) {
+      throw new SwapKitError("core_transaction_add_liquidity_invalid_params");
+    }
+    if (includeRuneAddress && !runeAddress) {
+      throw new SwapKitError("core_transaction_add_liquidity_no_rune_address");
+    }
+
+    const runeTx =
```

Comment:
You should probably do more than copy pasta code and change the naming. As there is not rune Transaction.

Reply:
This is playground function for the thorchain plugin ,not the mayachain.
Now I am working on the mayachain.